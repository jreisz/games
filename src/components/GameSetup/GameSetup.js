import React, { Component } from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import { connect } from "react-redux";
import {
  setDifficulty,
  setWidth,
  setHeight,
  setMines,
} from "../../store/SetUp/actions";
import { setRemainingFlags } from "../../store/MineSweeper/actions";
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  Row,
  Col,
} from "reactstrap";
import "./GameSetup.scss";

class GameSetup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dropdownOpen: false,
      dropDownValue: "Easy",
      boardHeight: 10,
      boardWidth: 10,
      boardMines: 10,
    };
    this.changeValue = this.changeValue.bind(this);
  }
  changeValue(e) {
    this.props.setDifficulty(e.currentTarget.textContent,e.currentTarget.id);
  }
  toggle() {
    this.setState({ dropdownOpen: !this.state.dropdownOpen });
  }
  render() {
    return (
      <div style={{ paddingTop: 50 }}>
        <Row>
          <Col>
            <span>
              Width:{" "}
              <input
                id="widthInput"
                readOnly="readOnly"
                className="__gameSetup_input-readonly"
                value={this.props.width}
              />{" "}
            </span>
            <span>
              Height:{" "}
              <input
                id="heightInput"
                value={this.props.height}
                className="__gameSetup_input-readonly"
                readOnly="readOnly"
              />{" "}
            </span>
            <span>
              Mines:{" "}
              <input
                id="minesInput"
                value={this.props.mines}
                className="__gameSetup_input-readonly"
                readOnly="readOnly"
              />
            </span>
          </Col>
        </Row>

        <Row>
          <br></br>
        </Row>
        <Row>
          <Col xs="12">
            <Dropdown
              isOpen={this.state.dropdownOpen}
              toggle={() => this.toggle()}
            >
              <DropdownToggle caret >
                Difficulty: {this.props.dropDownValue}
              </DropdownToggle>
              <DropdownMenu>
                <DropdownItem id="0" onClick={this.changeValue}>Kinder</DropdownItem>
                <DropdownItem id="1" onClick={this.changeValue}>Easy</DropdownItem>
                <DropdownItem id="2" onClick={this.changeValue}>Medium</DropdownItem>
                <DropdownItem id="3" onClick={this.changeValue}>Hard</DropdownItem>
              </DropdownMenu>
            </Dropdown>
          </Col>
        </Row>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    dropDownValue: state.Setup.difficulty,
    height: state.Setup.height,
    width: state.Setup.width,
    mines: state.Setup.mines,
  };
};
export default connect(mapStateToProps, {
  setDifficulty,
  setWidth,
  setHeight,
  setMines,
  setRemainingFlags,
})(GameSetup);
