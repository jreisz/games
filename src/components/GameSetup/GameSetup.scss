
.__gameSetup_input-readonly:read-only {
  background-color: rgb(247, 240, 240);
  opacity: 0.4;
  font-weight: bolder;
  width: 50px;
}
.dropdown button{
  color: #fff;
  background-color: #007bff;
  border-color: #007bff;
  
}
.dropdown button:hover {
  color: #fff;
  background-color: #0069d9;
  border-color: #0062cc;
}
.dropdown button:focus{
  color: #fff;
  background-color: #0069d9;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38,143,255,.5);
}
.dropdown button:not(:disabled):not(.disabled):active,.show>.btn-secondary.dropdown-toggle{
  color: #fff;
  background-color: #0069d9;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38,143,255,.5);
}
.btn-secondary:not(:disabled):not(.disabled).active:focus, .btn-secondary:not(:disabled):not(.disabled):active:focus, .show>.btn-secondary.dropdown-toggle:focus 
{
  box-shadow: 0 0 0 0.2rem rgba(38,143,255,.5); 
}
.dropdown-menu.show
  {
    transform: translate3d(712px, 36px, 0px)!important;
    border:0;
  }

.dropdown-menu.show button:first-of-type
{
  border-top-left-radius: 7px;
  border-top-right-radius: 7px;
}
.dropdown-menu.show button:last-of-type
{
  border-bottom-left-radius: 7px;
  border-bottom-right-radius: 7px;
}