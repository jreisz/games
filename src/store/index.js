import { createStore } from "redux";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import { combineReducers } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import Setup from "./SetUp/reducer";
import MineSweeper from "./MineSweeper/reducer";
import SavedGames from "./SavedGames/reducer";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(
  persistConfig,
  combineReducers({ Setup, MineSweeper, SavedGames })
);

export let store = createStore(persistedReducer, composeWithDevTools());
export  let persistor = persistStore(store);
