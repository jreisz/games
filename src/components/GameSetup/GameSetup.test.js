import React from 'react';
import ReactDOM from 'react-dom';
import GameSetup from './GameSetup';
import { Provider } from "react-redux";
import { store } from '../../store';
import { create } from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe("Game Setup", () => {
    let component;

    beforeAll(() => {
        component = mount(
            <Provider store={store}>
                <GameSetup />
            </Provider>
        );
    });
    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Provider store={store}>
                <GameSetup />
            </Provider>
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('should render with given state from Redux store', () => {
        const rendererComponent = create(<Provider store={store}>
            <GameSetup />
        </Provider>);
        expect(rendererComponent.toJSON()).toMatchSnapshot()
    });
    it('checks default difficulty', () => {
        const difficulty = component.find('button').first().text();
        expect(difficulty).toBe('Difficulty: ' + 'Kinder');
    });
    it('changes difficulty to Hard', () => {

        let difficulty = component.find('button').first().text();
        expect(difficulty).toBe('Difficulty: ' + 'Kinder');

        const optionHard = component.find('DropdownItem').at(3);
        optionHard.simulate('click', { });
        
        component.update()

        difficulty = component.find('button').first().text();
        expect(difficulty).toBe('Difficulty: ' + 'Hard');
        
    });    
});