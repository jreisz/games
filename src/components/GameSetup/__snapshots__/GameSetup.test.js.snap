// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Setup should render with given state from Redux store 1`] = `
<div
  style={
    Object {
      "paddingTop": 50,
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <span>
        Width:
         
        <input
          className="__gameSetup_input-readonly"
          id="widthInput"
          readOnly="readOnly"
          value={4}
        />
         
      </span>
      <span>
        Height:
         
        <input
          className="__gameSetup_input-readonly"
          id="heightInput"
          readOnly="readOnly"
          value={4}
        />
         
      </span>
      <span>
        Mines:
         
        <input
          className="__gameSetup_input-readonly"
          id="minesInput"
          readOnly="readOnly"
          value={1}
        />
      </span>
    </div>
  </div>
  <div
    className="row"
  >
    <br />
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          aria-label={null}
          className="dropdown-toggle btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Difficulty: 
          Kinder
        </button>
        <div
          aria-hidden={true}
          className="dropdown-menu"
          role="menu"
          tabIndex="-1"
        >
          <button
            className="dropdown-item"
            id="0"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            Kinder
          </button>
          <button
            className="dropdown-item"
            id="1"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            Easy
          </button>
          <button
            className="dropdown-item"
            id="2"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            Medium
          </button>
          <button
            className="dropdown-item"
            id="3"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            Hard
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
